pkgs/by-name/eu/euphonica/package.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cargo,
+  meson,
+  ninja,
+  pkg-config,
+  rustPlatform,
+  rustc,
+  wrapGAppsHook4,
+  cairo,
+  dbus,
+  gdk-pixbuf,
+  glib,
+  gtk4,
+  libadwaita,
+  openssl,
+  pango,
+  pipewire,
+  sqlite,
+  desktop-file-utils,
+  libxml2,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "euphonica";
+  version = "0.93.1-alpha";
+
+  src = fetchFromGitHub {
+    owner = "htkhiem";
+    repo = "euphonica";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-HfWjNoyAUd2fVRnc5Ii7suyTel7tPKjPs5bUmQCY6R4=";
+    fetchSubmodules = true;
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-fLjreVK/q+j71kBNdp7DT0uBoxgkB0fMVvxzRSgRPUc=";
+  };
+
+  nativeBuildInputs = [
+    cargo
+    meson
+    ninja
+    pkg-config
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+    rustc
+    wrapGAppsHook4
+    desktop-file-utils
+  ];
+
+  buildInputs = [
+    cairo
+    dbus
+    gdk-pixbuf
+    glib
+    gtk4
+    libadwaita
+    openssl
+    pango
+    pipewire
+    sqlite
+    libxml2
+  ];
+
+  meta = {
+    description = "MPD client with delusions of grandeur, made with Rust, GTK and Libadwaita";
+    homepage = "https://github.com/htkhiem/euphonica";
+    license = lib.licenses.gpl3Only;
```

```suggestion
    license = lib.licenses.gpl3Plus;
```
https://github.com/htkhiem/euphonica/blob/main/data/io.github.htkhiem.Euphonica.metainfo.xml.in#L14

Or you can also add cc0, but it's not important

---

Done!